### PR TITLE
Add "API catalog last updated" field to portability reports

### LINKED ISSUE #247: Include a timestamp of when the catalog was built on the reports
Right now, a report contains information about available APIs, but no notion of how old the data for that is. Since the available on .NET Core is continually changing, it would be helpful if the reports identified how old the catalog data was.

### PR DESCRIPTION
**Pull request by conniey**
Adding a build date to API catalog, so users know how up-to-date their results are.
Fixes #247.

@twsouthwick

### CODE REVIEW

**Comment by JJVertical** [comment]
This looks good to me

File: src/Microsoft.Fx.Portability/ObjectModel/CloudApiCatalogLookup.cs [hunk 1]
```diff
@@ -211,6 +213,8 @@ public IEnumerable<string> GetAncestors(string docId)
             }
         }
 
+        public DateTimeOffset BuiltOn { get { return _builtOn; } }
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability/ObjectModel/CloudApiCatalogLookup.cs]: The name "BuiltOn" seems a little ambiguous - built on what? How about 'LastModified' or something?

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml [hunk 2]
```diff
@@ -15,6 +15,7 @@
 @{
     ViewBag.Title = LocalizedStrings.HtmlReportTitle;
     var submissionId = Model.ReportingResult.SubmissionId;
+    var catalogLastUpdated = Model.CatalogBuiltOn.ToString("D", System.Globalization.CultureInfo.InvariantCulture);
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml]: Why invariant? This should be dependent on the current culture

File: src/Microsoft.Fx.Portability/ObjectModel/CloudApiCatalogLookup.cs (hunk 1 above)
Comment (conniey) [inline src/Microsoft.Fx.Portability/ObjectModel/CloudApiCatalogLookup.cs]: Fixed

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml (hunk 2 above)
Comment (conniey) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml]: Fixed

**Comment by conniey** [comment]
@twsouthwick : Does this look OK?

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml [hunk 3]
```diff
@@ -15,6 +15,7 @@
 @{
     ViewBag.Title = LocalizedStrings.HtmlReportTitle;
     var submissionId = Model.ReportingResult.SubmissionId;
+    var catalogLastUpdated = Model.CatalogLastUpdated.ToString("D", Model.UserCulture);
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml]: Can you not just access System.Globalization.CultureInfo.CurrentCulture?

Comment (conniey) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/ReportTemplate.cshtml]: True. I overthought it.

**Comment by twsouthwick** [comment]
LGTM

**Pull request merged**